core/templates/pages/topic-editor-page/modal-templates/create-new-story-modal.controller.spec.ts

```diff
@@ -84,16 +84,16 @@ describe('Create New Story Modal Controller', function() {
       StoryEditorStateService,
       'getStoryWithUrlFragmentExists').and.returnValue(true);
     expect($scope.storyUrlFragmentExists).toBeFalse();
-    $scope.story.urlFragment = 'test-url';
-    $scope.onStoryUrlFragmentChange();
+    $scope.unsubmittedUrlFragment = 'test-url';
+    $scope.assignUrlFragment();
```

Comment:
Why this change?

Reply:
And ditto the stuff below.

Reply:
@vojtechjelinek, I had added a button for submitting the url and as a part of this I had removed the onStoryUrlFragmentChange() function and moved it functionality to assignUrlFragment() function which is called when the button is clicked. The unsubmittedUrlFragment is the temporary variable that holds the url fragment before it is submitted. I changed the frontend tests here accordingly.

Would this be fine?

Reply:
Umm I don't think this is the right solution to the problem. Also if we were to do it this way we would need to do it in all places where we have a similar code.

Reply:
@vojtechjelinek, I didnt really understand, is it that the frontend tests are not correct? (as in, it wont work) or is it that the coding style followed by me should be changed? Would you please give me some clarity into the issue.

Reply:
> Umm I don't think this is the right solution to the problem. Also if we were to do it this way we would need to do it in all places where we have a similar code.

Do you mean adding the button is not the right solution to the problem? It was added because the end to end tests would otherwise fail. The hyphen in the url fragment gives error as the story editor fills it in. What would you suggest to be an alternate solution?

Reply:
I think the schema changes should not break the e2e tests, you should find as solution that works without introducing the button. Either modify the e2e tests or modifying the schema.

Reply:
@vojtechjelinek, would there be other files that must be changed? Since story creation is present only in `core/templates/pages/topic-editor-page/modal-templates/create-new-story-modal.template.html` right.

Reply:
@vojtechjelinek, a possible solution would be to remove the hyphens from the e2e tests, for example in the file `core/tests/protractor_desktop/learnerDashboard.js`, theres a test with url fragment "story-one" which if changed to "storyone" would allow the e2e tests to pass. 

Otherwise, the regex in the schema that checks the url fragment can be modified to allow url fragments that end with a hyphen to pass. 

Would either of these be viable solutions?

Reply:
Also the thing is that, I think story creation is present in `core/templates/pages/topic-editor-page/modal-templates/create-new-story-modal.template.html` and not in any other file (the `onStoryUrlFragmentChange()` function also exists only in the `core/templates/pages/topic-editor-page/modal-templates/create-new-story-modal.controller.ts` file) as well as all the frontend, e2e and backend tests passed. 
So would there be any other file where changes have to be made? So we can consider the submit button as a fair solution right.

Reply:
> @vojtechjelinek, a possible solution would be to remove the hyphens from the e2e tests, for example in the file `core/tests/protractor_desktop/learnerDashboard.js`, theres a test with url fragment "story-one" which if changed to "storyone" would allow the e2e tests to pass.
> 
> Otherwise, the regex in the schema that checks the url fragment can be modified to allow url fragments that end with a hyphen to pass.
> 
> Would either of these be viable solutions?

Yes, removing the hyphens seems fine.



> Also the thing is that, I think story creation is present in `core/templates/pages/topic-editor-page/modal-templates/create-new-story-modal.template.html` alone and not in any other file (the `onStoryUrlFragmentChange()` function also exists only in the `core/templates/pages/topic-editor-page/modal-templates/create-new-story-modal.controller.ts` file alone) as well as all the frontend, e2e and backend tests passed. So would there be any other file where changes have to be made? So we can consider the submit button as a fair solution right.

The problem is that topic or skill creation are quite similar to story creation and I think they should be pretty similar, but now we only use the button in the story creator.

Reply:
>The problem is that topic or skill creation are quite similar to story creation and I think they should be pretty similar, but now we only use the button in the story creator.
Oh I see
>Yes, removing the hyphens seems fine.
Thanks, I will implement that as soon as possible